nibabel/tests/test_diff.py

```diff
@@ -7,9 +7,13 @@
 from os.path import (dirname, join as pjoin, abspath)
 import numpy as np
 
-from hypothesis import given
-import hypothesis.strategies as st
-
+try:
+    import hypothesis
+    from hypothesis import given
+    import hypothesis.strategies as st
+except Exception as exc:
+    from nose import SkipTest
+    raise SkipTest("Failed to import hypothesis or its components: %s" % exc)
```

Comment:
To avoid necessarily skipping the entire file, what about making this `except` statement:

```Python
except ImportError:
    class st(object):
        @classmethod
        def data(klass):
            return None

    def given(obj):
        def decorator(func):
            def null_test():
                return None
        return decorator
```

Maybe raise a warning, as well.